doc/whats-new.rst

```diff
@@ -21,7 +21,9 @@ v0.15.0 (unreleased)
 
 Breaking changes
 ~~~~~~~~~~~~~~~~
-
+- :py:func:`xarray.dot`, :py:meth:`DataArray.dot`, and the ``@`` operator now
+  use ``align="inner"`` (except when ``xarray.set_options(arithmetic_join="exact")``;
+  :issue:`3694`) by `Mathias Hauser <https://github.com/mathause>`_.
```

Comment:
It stops raising an error when your data does not align, no change if it does; so I assume it is unlikely for anyone to rely on this raising?